From d37402f72dbf151378ffd28ca60ff1b691afd4df Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 18 Oct 2021 18:41:25 +0300
Subject: [PATCH] [SW-57369] gaudi2: agent: set preboot version

Currently, the preboot version is set from BTL based preboot compilation
for both agent fits (BTL and secureboot). This patch add change to
set preboot version compiled as part of secureboot for secureboot
agent fit and set preboot version of preboot compiled as part of BTL
flow for BTL agent fit.

Change-Id: I2d0493e0ffd3413f1b75b3500c5c1958750e542d
---

diff --git a/board/habanalabs/gaudi2/post-image-agents.sh b/board/habanalabs/gaudi2/post-image-agents.sh
index 4edcb2e..3a3025c 100755
--- a/board/habanalabs/gaudi2/post-image-agents.sh
+++ b/board/habanalabs/gaudi2/post-image-agents.sh
@@ -7,6 +7,9 @@
 ASIC_SRAM_BASE=0x0000007F
 ASIC_OFFSET_ADD=0xFD000000
 
+# Set hl-preboot-version = "hl-<device>-<HL_VERSION>[-rc]-fw-<fw ver>[-<git data>] (<date> - <time>)"
+HL_PREBOOT_VERSION=`cat ${BR2_EXTERNAL}/output/${DEV_PREFIX}_arc/build/preboot-custom/buildversion-habanalabs`
+
 function handle_fw_copy {
 
 	# Copy common arc files required for agent fit creation
@@ -40,6 +43,8 @@
 	AGENT_ITB_FILE_LEGACY=agent-${AGENT_NAME}-fit_legacy.itb
 	AGENT_ITS_TEMPLATE_LEGACY=${BOARD_BASE}/agent-${AGENT_NAME}_legacy.its
 	sed ${ITS_SED_PARAMS}${ITS_SED_BIN_PARAMS} ${AGENT_ITS_TEMPLATE_LEGACY} > ${AGENT_ITS_FILE_LEGACY}
+	# Set preboot version from btl base preboot
+	HL_PREBOOT_VERSION=`cat ${BASE_DIR}/build/preboot-custom/buildversion-habanalabs`
 	${HOST_DIR}/usr/bin/mkimage -D "-W no-unit_address_vs_reg" -f ${AGENT_ITS_FILE_LEGACY} -k ${KEYS_DIR} ${AGENT_ITB_FILE_LEGACY}
 
 	# KMD like this to be 4B aligned in size
